src/services/jsTyping.ts

```diff
@@ -46,8 +55,44 @@ namespace ts.JsTyping {
             }
         }
         
-        let newTypingNames = typingNames.filter(t => cachedTypingNames.indexOf(t) < 0);
-        
+        let newTypingNames = ts.filter(typingNames, t => cachedTypingNames.indexOf(t) < 0);
         return { cachedTypingPaths, newTypings: newTypingNames };
     }
+
+    export function updateTypingsConfig(
+        host: TypeDefinitionResolutionHost | System, cachedTypingsPaths: string[], newTypings: string[], cachePath: string, typingsConfigPath: string): void {
+        let tsdJsonCacheInfo = tryGetTsdJsonCacheInfo(host, cachePath);
+        if (tsdJsonCacheInfo) {
+            let notFound: string[] = [];
+            let cacheTsdJsonDict = tsdJsonCacheInfo.value;
+            if (cacheTsdJsonDict.hasOwnProperty("installed")) {
+                let installedTypings = Object.keys(cacheTsdJsonDict.installed);
+                notFound = ts.filter(newTypings, i => !isInstalled(i, installedTypings));
+            }
+            if (cacheTsdJsonDict.hasOwnProperty("notFound")) {
+                notFound = ts.deduplicate<string>(cacheTsdJsonDict["notFound"].concat(notFound));

```

Comment:
Also this may not be true. What used to be notFound may be found as the repo gets updated constantly.